### PR TITLE
Admin API Keys

### PR DESCRIPTION
**Pull request by mikehelmick**
* Introduce Admin API keys, that will be used for issuing diagnosis codes
* Existing keys auto migrated to be device keys
* UI for creating these

### CODE REVIEW

File: pkg/database/authorized_app.go [hunk 1]
```diff
@@ -29,10 +29,14 @@ const (
...
+//
+// Admin Keys are able to issue diagnosis keys and are not able to perticipate
+// the verification protocol.
 type AuthorizedApp struct {
 	gorm.Model
-	Name   string `gorm:"type:varchar(100);unique_index"`
-	APIKey string `gorm:"type:varchar(100);unique_index"`
+	Name     string `gorm:"type:varchar(100);unique_index"`
+	APIKey   string `gorm:"type:varchar(100);unique_index"`
+	AdminKey bool   `gorm:"default:false"`
```
Comment (sethvargo) [inline pkg/database/authorized_app.go:39]: Should we consider making this an enum (e.g. KeyType)? The boolean feels weird to me.

**Comment by sherifkozman** [comment]
@mikehelmick will that enable external integration to use API calls to create verification codes without having to relay on the UI ?

File: pkg/database/authorized_app.go (hunk 1 above)
Comment (mikehelmick) [reply pkg/database/authorized_app.go]: changed to int / enum - right now I don't see the need for more than 2 key types, but I'm sure I'll be wrong tomorrow :)

**Comment by mikehelmick** [comment]
> @mikehelmick will that enable external integration to use API calls to create verification codes without having to relay on the UI ?

correct.

File: pkg/database/authorized_app.go [hunk 2]
```diff
@@ -24,6 +24,9 @@ import (
 
 const (
 	apiKeyBytes = 64 // 64 bytes is 86 chararacters in non-padded base64.
+
+	APIUserTypeDevice = 0
```
Comment (sethvargo) [inline pkg/database/authorized_app.go:28]: iota?

Comment (sethvargo) [reply pkg/database/authorized_app.go:28]: and should this be a custom type?

Comment (mikehelmick) [reply pkg/database/authorized_app.go]: made it a custom type , plumbed all the way through

**Comment by mikehelmick** [comment]
Tagging issue #95

**Pull request merged**